std/utf.d

```diff
@@ -1164,8 +1164,10 @@ package template codeUnitLimit(S)
  * Returns:
  *      decoded character
  */
-private dchar decodeImpl(bool canIndex, UseReplacementDchar useReplacementDchar = UseReplacementDchar.no, S)(auto ref S str, ref size_t index)
-    if (is(S : const char[]) || (isInputRange!S && is(Unqual!(ElementEncodingType!S) == char)))
+private RV decodeImpl(bool canIndex, UseReplacementDchar useReplacementDchar, RV = dchar, S)
```

Comment:
@burner Okay, rather than the `RV` bit, it now takes a new enum type, `DecodeType` and returns `auto`, which makes more logical sense, though it does require a new, private enum type. But `UseReplacementDchar` casts to it nicely (tests for that are included), so it's not too disruptive to the various calls to `decodeImpl`. This is probably what I should have done in the first place.